### ISSUE TITLE
Cannot Load Custom Resources in Pack (pck) files

### ISSUE THREAD

**Issue by pondc1** [godotengine/godot#72051]
### Godot version

4.0-beta15

### System information

Windows 11

### Issue description

In the first project `TestExport` I have a custom resource `TestResource` that contains an export for a String and int variables, as well as another script that contains an export for said custom resource that prints out the String and int vars on `_ready`. This runs fine in its own project, which I then export as a pack file called `MyPack.pck`. I then have another project `TestImport` which contains a script to load `MyPack.pck` and loads the scene in `TestExport`, this should print out the contents of `TestResource` but results in the following error when running: `Parser Error: "TestResource" was not found in the current scope.` 

Can we not have custom resources in pack files?

### Steps to reproduce

Getting the Pack File (optional if you want to reexport the pack file):

1. Open the `TestExport` project
2. Go to the top and do Project->Export...
3. In the export dialog, click on the `Windows Desktop` template on the left
4. At the bottom of the dialog, press `Export PCK/ZIP...`
5. Name the pack file `MyPack.pck`
6. Copy the pack file into the other project `TestImport/PackFiles`

Hitting the error

1. Open the `TestImport` project
2. Verify that `MyPack.pck` exists in `TestImport/PackFiles`
3. Run the project (F5)

### Minimal reproduction project

[TestExport.zip](https://github.com/godotengine/godot/files/10500978/TestExport.zip)
[TestImport.zip](https://github.com/godotengine/godot/files/10500980/TestImport.zip)

**Comment by pondc1** [comment]
Gonna go ahead and bump this. This is still happening on the new 4.0 RC1. However, this time there is an extra oddity. If you create a new custom resource (or if you were to load the `TestExport` project into a new godot version such as RC1), the project will do one of two things:

1.  It will report the same `"TestResource" was not found in the current scope.` until the project is restarted.
2. It will let you create a new `TestResource` object but wont let you assign it in the inspector until the project is restarted

Regardless, it ultimately still results in the initial error in the `TestImport` project when attempting to load it via PCK file

**Comment by pondc1** [comment]
Bumping this again as it is still happening in 4.0-stable.

**Comment by dpalais** [comment]
If you remove the imported script class_name, it works as expected. I believe the issue with the test project you provided is that the host godot project, the importer, doesn't merge the script classes list from the imported pack into the running system. Thus it doesn't know what a MyClassName is and understandably fails. But you can just attach any script to a resource or node without the use of class_name.

Here's your test project modifed to work. Tested in Godot 4.2 RC.
[Test.zip](https://github.com/godotengine/godot/files/13468822/Test.zip)

**Comment by dpalais** [comment]
Related: #61556
Related: #82061

**Comment by pondc1** [comment]
Yes I am aware that you you use the generic resource, all the data is passed with it and it is a _solution_. But we were looking for something that is more type safe where we can use the defined `class_name` and that seems not possible if it comes from a pck file.
